### DISCUSSION TITLE
Working on DBeaver

### DISCUSSION THREAD

**Discussion by MaikVink** [dbeaver/dbeaver#16156] [DBeaver Development]
Hi,
I have been trying to figure out how DBeaver works.
Now, I'm not an expert with Java so that is maybe not very helpful, but I'm working on that part from my side (learning how to write code in Java), but so far it doesn't help me finding out how certain function work. It seems that some pieces of code are even never used or outdated some times.

Is there any guide that describes how the code works (and/or maybe how you can work with the code)?

An other option would also be that I can have a conversation with one/some of the members of DBeaver who know how to work with the code, to who I could ask a couple of questions. Would this by any chance this could be possible?

If there is any other possibility to help me understand the source code of DBeaver (apart for trying to figure it out myself or to just follow Java guides), you are welcome to give suggestions.

thanks in advance

**Comment by LonwoLonwo** [comment]
Hello @MaikVink 

You can ask your questions here. Or in the new discussion. With "development" label, as you did it for this question. 

All development info you can find here https://github.com/dbeaver/dbeaver/wiki in the "Development" section.
